Modules/_json.c

```diff
@@ -1667,12 +1719,20 @@ encoder_listencode_list(PyEncoderObject *s, _PyUnicodeWriter *writer,
         Py_CLEAR(ident);
     }
 
-    /* TODO: DOES NOT RUN
     if (s->indent != Py_None) {
-        indent_level -= 1;
+        indent_level--;
+        Py_DECREF(separator);
+        newline_indent = _create_newline_indent(s->indent, indent_level);
```

Comment:
`newline_indent` is calculated twice per every list or dict. First for nested items, second for the closing `]` or `}`. The latter is the same as `newline_indent` for items of the outer list or dict.

In the following example the indent for the nested `]` is the same as the indent for `1`, `2` and the nested `[`:
```
[
    1,
    2,
    [
        3,
        4
    ]
]
```
I think that we can just pass it as an argument of `encoder_listencode_list()` and `encoder_listencode_dict()`.

The inner value can then be calculated as a simple concatenation, without multiplication:
```c
item_newline_indent = PyUnicode_Concat(newline_indent, s->ident);
```